numpy/core/src/multiarray/descriptor.c

```diff
@@ -1859,72 +1859,17 @@ arraydescr_protocol_typestr_get(PyArray_Descr *self)
 }
 
 static PyObject *
-arraydescr_typename_get(PyArray_Descr *self)
+arraydescr_name_get(PyArray_Descr *self)
 {
-    static const char np_prefix[] = "numpy.";
-    const int np_prefix_len = sizeof(np_prefix) - 1;
-    PyTypeObject *typeobj = self->typeobj;
+    /* let python handle this */
+    PyObject *_numpy_dtype;
```

Comment:
Shouldn't this be `static PyObject *_numpy_dtype = NULL;`  It needs to stay around and needs to be initialized.

Reply:
I don't think it does need to stay around, and it's initialized at its first use below.

This matches `dtype.__repr__`, in that both do not bother to cache the module or the function